cmd/gcp-controller-manager/node_csr_approver_test.go

```diff
@@ -1640,3 +1644,126 @@ func TestParseInstanceGroupURL(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteAllPodsBoundToNode(t *testing.T) {
+	testNode := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{Name: "testNode"},
+		Status: v1.NodeStatus{
+			Capacity: v1.ResourceList{
+				v1.ResourceName(v1.ResourceCPU):    resource.MustParse("10"),
+				v1.ResourceName(v1.ResourceMemory): resource.MustParse("10G"),
+			},
+		},
+	}
+
+	testPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "default",
+			Name:      "testPod",
+		},
+		Spec: v1.PodSpec{
+			NodeName: "testNode",
+		},
+		Status: v1.PodStatus{
+			Conditions: []v1.PodCondition{
+				{
+					Type:   v1.PodReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+	}
+
+	nodeList := &v1.NodeList{}
+	nodeList.Items = append(nodeList.Items, *testNode)
+	podList := &v1.PodList{}
+	podList.Items = append(podList.Items, *testPod)
+
+	client := fake.NewSimpleClientset(nodeList, podList)
+	controllerCtx := &controllerContext{client: client}
+
+	err := deleteAllPodsBoundToNode(controllerCtx, "testNode")
+
+	if err != nil {
+		t.Fatalf("unexpected error deleting all pods bound to node %v", err)
+	}
+
+	actions := client.Actions()
+	if len(actions) != 3 {
+		t.Fatalf("unexpected number of actions, got %v, want 3 (list pods, patch pod, delete pod)", len(actions))
+	}
+
+	var didFindPatchPodAction bool
+	var didFindDeletePodAction bool
+
+	for _, action := range actions {
+		if action.Matches("patch", "pods") {
+			didFindPatchPodAction = true
+
+			scheme := runtime.NewScheme()
+			codecs := serializer.NewCodecFactory(scheme)
+			_ = v1.AddToScheme(scheme)
+
+			patchedPodBytes := action.(testclient.PatchAction).GetPatch()
+			obj, _, err := codecs.UniversalDeserializer().Decode(patchedPodBytes, nil, nil)
+			if err != nil {
+				t.Fatalf("failed to decode pod patched bytes: %v", err)
+			}
+
+			patchedPod, ok := obj.(*v1.Pod)
+			if !ok {
+				t.Fatalf("unexpected object, got: %#v expected: v1.Pod", obj)
```

Comment:
nit: some error lines contain `expected:` and some `want:`. Suggest to use `want` consistently.

Reply:
Updated to be a bit more consistent.